src/ui/components/results/verticalresultscomponent.js

```diff
@@ -243,6 +258,53 @@ export default class VerticalResultsComponent extends Component {
     return 'results/verticalresults';
   }
 
+  /**
+   * Given an array of nlp filters from the backend (see class AppliedQueryFilter within section.js),
+   * turn them into an array of SimpleFilterNodes
+   * @param {Array<Object>} nlpFilters
+   * @returns {Array<SimpleFilterNode>}
+   */
+  _convertNlpFiltersToFilterNodes (nlpFilters) {
+    return nlpFilters.map(nlpFilter => FilterNodeFactory.from({
+      filter: nlpFilter.filter, metadata: nlpFilter.metadata
+    }));
+  }
+
+  /**
+   * Recursively get all of the root FilterNodes of the given array,
+   * which will all be SimpleFilterNodes
+   * @param {Array<FilterNode>} filterNodes
+   * @returns {Array<SimpleFilterNode>}
+   */
+  _getSimpleFilterNodes (filterNodes) {
```

Comment:
I'm wondering if there's a better place for this method. Conceivably, we could use it somewhere else. It feels more like an instance method we should add to the FilterNode class. What if we add a getSimpleAncestors() to the FilterNode class? It returns all SimpleFilterNode ancestors (at any level) of the FilterNode.

Reply:
Aw man I remember trying to push for this with FilterView but I guess it makes a lot more sense on a node class, added!